src/App.tsx

```diff
@@ -35,61 +35,136 @@ export function getReorderedGoods(
   const visibleGoods = [...goods];
 
   // Sort and reverse goods if needed
+  if (sortType === SortType.NONE && isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  if (sortType === SortType.ALPHABET) {
+    return !isReversed
+      ? visibleGoods.sort()
+      : visibleGoods.sort((a, b) => (a > b ? -1 : 1));
+  }
+
+  if (sortType === SortType.LENGTH) {
+    return !isReversed
+      ? visibleGoods.sort((a, b) => a.length - b.length)
+      : visibleGoods.sort((a, b) => a.length - b.length).reverse();
+  }
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+// eslint-disable-next-line react/prefer-stateless-function
+export default class App extends React.Component {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      ...state,
+      isReversed: !this.state.isReversed,
```

Comment:
Try to fix it
<img width="934" alt="image" src="https://user-images.githubusercontent.com/99403198/222891055-7f67c90f-b8ad-4404-94af-7ef3a8da1c44.png">
```suggestion
      isReversed: !state.isReversed,
```

Reply:
@Anastasiia-Svintsova https://mate-parttime.slack.com/archives/C041JA5EJKV/p1677939513973379